src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -277,19 +277,24 @@ parseLocalDeclaration = positioned (P.choice
 parseModule :: TokenParser Module
 parseModule = do
   comments <- C.readComments
+  start <- P.getPosition
```

Comment:
We do, but it doesn't quite work for this case, it's for `PositionedX` style constructors. I considered doing something like:

``` haskell
withSourceSpan Module $ do
  ...
  return ($ name decls exports)
```

But that doesn't work either without changing `withSourceSpan`.